### DISCUSSION TITLE
How the projects' new versions identified?

### DISCUSSION THREAD

**Discussion by lmilbaum** [fedora-infra/anitya#1648] [Q&A]
I'm interested in the source code that handles the identification. For example, the following project: https://release-monitoring.org/project/250162/
BTW, where the data is tracked?

**Comment by lmilbaum** [comment]
Another question. Where this project is used and for what purpose?

**Comment by Zlopez** [comment]
This project was obtained by automated consumer for libraries.io, the code for the consumer is [here](https://github.com/fedora-infra/anitya/blob/master/anitya/librariesio_consumer.py)

**Reply by lmilbaum** [reply]
Not sure I follow. I'm asking those questions because Renovate is doing similar things. Was hoping to find out if it can be used in someway in this project.

**Reply by Zlopez** [reply]
We are already using Renovate for dependencies, see https://github.com/fedora-infra/anitya/issues/1458.

Maybe I didn't understood your questions correctly. Were you asking about the project https://release-monitoring.org/project/250162/ or about the Anitya itself?

**Reply by lmilbaum** [reply]
I'll try to clarify it in a different way. 
If I understand correctly, Anitya is scraping packages data and publish it (website, messages on message bus).
Renovate is scraping packages version data and publish it via PRs.

Sounds to me like a good basis for utilizing Renovate for the scraping packages data part.

**Comment by Zlopez** [comment]
That sounds like an interesting idea, it would be probably interesting to look at how renovate is doing that.

**Reply by lmilbaum** [reply]
Agreed. How can we push this forward?

**Comment by Zlopez** [comment]
The best start would be to check if they have any API that could be used or just look how they implement things.

**Reply by lmilbaum** [reply]
I'm not sure they have APIs. It is a service (self-managed or GitHub app) which polls the data on a regular cadence. Eventually, it generates PRs/MRs with the updates.  
Would it make sense to track all the packages with their pinned versions in this project and let Renovate generate those update PRs here?

**Comment by Zlopez** [comment]
But this is already happening, Anitya is using Renovate for tracking all the dependencies for it.

**Reply by lmilbaum** [reply]
I'm not sure I explained my idea clearly enough.  Maybe I'll ask a question first.
How is Anitya identifies that a new version is available for the following project https://release-monitoring.org/project/28569/ ?

**Reply by Zlopez** [reply]
So this is done based on the specified backend (in this case PyPI). The check is right now going like this:
1) Get all the projects eligible for check (not checked in last hour, not on the blacklisted backend - this is currently used only for GitHub ratelimiting)
2) Run the check based on the backend (plenty of projects are just checking the HTML for correct regex, some are using API or specific endpoint - JSON endpoint in case of PyPI)
3) If any new version is obtained, save it to project
4) Wait 5 minutes and repeat the process from step 1

**Reply by lmilbaum** [reply]
Don't you think Renovate can do that as well?
 Let say you have a text file in anitya project with the following content:
`mercurial  6.5`.
Renovate can identify the new version (when/if available) and update the file content via PR.
Merging the PR will save updated data to the project.

**Reply by Zlopez** [reply]
This will be much more resource heavy processing PRs that are created against some repository (not sure which one). It doesn't seem better and it will definitely not support all the backends Anitya has.
It would be much better to just investigate if they are using something smart when obtaining the versions instead of just checking projects one by one.

**Reply by lmilbaum** [reply]
The discussion becomes very technical. I think it worth an hour of our time to rehash it.
Can we hop on a call some time soon?

**Reply by Zlopez** [reply]
Yes, we can. What time do you propose and where should we meet?

**Reply by lmilbaum** [reply]
Please reach out over Red Hat Slack workspace.

**Reply by Zlopez** [reply]
OK

**Reply by lachmanfrantisek** [reply]
@Zlopez @lmilbaum Liora, Michal, just so you know, you both are invited to the meeting tomorrow where we can discuss this..;-D

**Reply by lachmanfrantisek** [reply]
Actually, this was one of the reasons to organise the call, to get you two together..;)
